### PR TITLE
Phgrote/admin create course

### PR DESCRIPTION
**Pull request by phgrote**

### CODE REVIEW

File: http/courseHandler.go [hunk 1]
```diff
@@ -85,3 +87,59 @@ func (a *AppServer) GetCourseHandler() httprouter.Handle {
 		}
 	}
 }
+
+func (a *AppServer) CreateCourseHandler() httprouter.Handle {
+	type request struct {
```
Comment (ldb) [inline http/courseHandler.go:178]: `request` and `response` don't need `bson` tags here.

File: http/courseHandler.go [hunk 2]
```diff
@@ -85,3 +87,59 @@ func (a *AppServer) GetCourseHandler() httprouter.Handle {
...
+		err := json.NewDecoder(r.Body).Decode(&request)
+		if err != nil {
+			w.WriteHeader(http.StatusBadRequest)
+			return
+		}
+
+		course.Title = request.Title
+		course.Description = request.Description
+		course.Members = request.Members
+		course.Members = request.Labels
```
Comment (ldb) [inline http/courseHandler.go]: `course.Members` should be `course.Labels`.

File: http/courseHandler.go [hunk 3]
```diff
@@ -85,3 +87,59 @@ func (a *AppServer) GetCourseHandler() httprouter.Handle {
...
+		Members     []string `json:"members,omitempty" bson:"members,omitempty"`
+		Labels      []string `json:"labels" bson:"labels"`
+	}
+	type response struct {
+		ID          bson.ObjectId          `json:"id,omitempty" bson:"_id"`
+		Title       string                 `json:"title,omitempty" bson:"title,omitempty"`
+		Description string                 `json:"description,omitempty" bson:"description,omitempty"`
+		Members     []string               `json:"members,omitempty" bson:"members,omitempty"`
+		Labels      []string               `json:"labels" bson:"labels"`
+		EntryIDs    []bson.ObjectId        `json:"entryIDs" bson:"entryIDs"`
```
Comment (ldb) [inline http/courseHandler.go]: `EntryIDs`should not be published I think.

File: http/middleware.go [hunk 4]
```diff
@@ -36,6 +37,29 @@ func NewAuthMiddleware(provider eduboard.UserAuthenticationProvider) func(handle
 	}
 }
 
+func BasicAuth(username, password string) func(handler http.Handler) http.Handler {
```
Comment (ldb) [inline http/middleware.go]: Let's discuss in the group but I don't understand why we need a different authentication for admins here.

File: http/routes.go [hunk 5]
```diff
@@ -32,3 +32,9 @@ func (a *AppServer) publicRoutes() *httprouter.Router {
 	router.POST("/api/logout", a.LogoutUserHandler())
 	return router
 }
+
+func (a *AppServer) adminRoutes() *httprouter.Router {
+	router := httprouter.New()
+	router.POST("/api/v1/admin/courses", a.CreateCourseHandler())
+	return router
+}
```
Comment (cosmonawt) [inline http/routes.go]: Newline for GitHub 😊

File: service/courseService/courseService.go [hunk 6]
```diff
@@ -37,3 +38,12 @@ func (cS CourseService) GetCourse(id string, cef eduboard.CourseEntryManyFinder)
...
 	return nil, course
 }
+
+func (cS CourseService) CreateCourse(c *eduboard.Course) (*eduboard.Course, error) {
+	err := cS.CR.Insert(c)
+	if err != nil {
+		return &eduboard.Course{}, errors.Wrap(err, "error storing user")
+	}
+	return c, nil
+
```
Comment (ldb) [inline service/courseService/courseService.go]: Empty line

File: service/courseService/courseService.go [hunk 7]
```diff
@@ -37,3 +38,12 @@ func (cS CourseService) GetCourse(id string, cef eduboard.CourseEntryManyFinder)
...
+}
```
Comment (ldb) [inline service/courseService/courseService.go]: Newline for GitHub 😊

File: http/middleware.go (hunk 4 above)
Comment (ldb) [reply http/middleware.go]: Clarified. Admins are treated independently from teachers and students, thus get a different auth method as well. 👍

**Review by ldb — APPROVED** [review]

**Pull request merged**